### ISSUE TITLE
Share link not visible because of ad blockers

### ISSUE THREAD

**Issue by TecJon** [nextcloud/server#10434]
### Steps to reproduce
1. Click on share symbol for file or folder
2. Check out side panel for sharing
3. Miss share link option

### Expected behaviour
I expect an option to share a link.

### Actual behaviour
Only input for user sharing is visible
![capture](https://user-images.githubusercontent.com/11375752/43356066-10c6764e-926a-11e8-862b-18eb9a76a350.PNG)

### Server configuration

**Nextcloud version:** 14.0.0 Beta 1

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh

**Where did you install Nextcloud from:**

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 61.0.1 (64-bit)

**Operating system:** Win 10

#### Browser log
<details>
<summary>Browser log</summary>

```
Content Security Policy: Directive ‘child-src’ has been deprecated. Please use directive ‘worker-src’ to control workers, or directive ‘frame-src’ to control frames respectively.
JQMIGRATE: Migrate is installed, version 1.4.0 core.js:7:542
Content Security Policy: The page’s settings blocked the loading of a resource at https://cloud.example.com/index.php/apps/files/ (“script-src”).
<script> source URI is not allowed in this document: “https://cloud.example.com/apps/notifications/js/notifications.js?v=1f8ad6d7-0”. files:51
unreachable code after return statement[Learn More] merged.js:1494:4
window.controllers/Controllers is deprecated. Do not use it for UA detection. merged.js:2171
New search handler registered search.js:66:4
querying Whats New data was successful: nocontent merged-template-prepend.js:4455:4
undefined  merged-template-prepend.js:4456:4
```
</details>

**Comment by rullzer** [comment]
@nextcloud/javascript seems like some glitch...

**Comment by skjnldsv** [comment]
Wut!

**Comment by skjnldsv** [comment]
I cannot reproduce: 
![capture d ecran_2018-07-30_12-57-56](https://user-images.githubusercontent.com/14975046/43393759-77f40176-93f8-11e8-9bc0-48508c135e90.png)

Browser: Firefox 61.0.1 (64-bit)

@TecJon can you clear your caches?

**Comment by TecJon** [comment]
@skjnldsv uBlock Origin was evildoer. When enabled it will not show up (even though reportedly 0% of the page is blocked). Might still be worth looking into?

**Comment by skjnldsv** [comment]
@TecJon What is the filter blocking this? A custom one?
Because I also have ublock origin (default settings) and it still shows :)

**Comment by TecJon** [comment]
It seems like the option "Parse and enforce cosmetic filters" is the reason. I'm guessing this filter is not enabled in your environment, @skjnldsv ?

> This option enables the parsing and enforcing of Adblock Plus-compatible “element hiding” filters. These filters are essentially cosmetic, they serve to hide elements in a web page which are deemed to be a visual nuisance, and which can't be blocked by the net request-based filtering engine. 

I guess that is why it says 0% is blocked. It's not a net request-based filtering.

**Comment by skjnldsv** [comment]
I do not have this option available, where do you find this?

**Comment by TecJon** [comment]
I find it in the filter lists tab of the ublock origin settings. 
See this screenshot from their github page. Mine looks slightly different and the tabs name is "Filter lists" instead of "3rd-party filters" but the location is the same.
I know I added some filters which were not selected by default. I do not use any custom filters.

![filter](https://cloud.githubusercontent.com/assets/585534/24972255/cbf77306-1f88-11e7-808e-1e1ac934120f.png)

**Comment by skjnldsv** [comment]
Nope, I can't reproduce. All of my settings are already enabled and the parse and enforce cosmetic filters too. :/

@nextcloud/designers can one of you reproduce?

**Comment by blizzz** [comment]
Cannot reproduce either with uBock origin enabled.

**Comment by skjnldsv** [comment]
Sorry @TecJon 
Thanks a lot for reporting, let's close this one and see if people complains as well (you're the only one so far)
Let's hope this is something unrelated :)

**Comment by TecJon** [comment]
Will try to find out which filter it was. Probably some odd combination.

**Comment by tplusm** [comment]
Running into the same issue with Adguard. When enabled, the 3 dots (...) to copy public link do not show. Turn Adguard off, 3 dots appear, clicking on it shows an option to copy public link. 
Using basic filters on Adgaurd.

**Comment by skjnldsv** [comment]
Dammit :grin: 
Can someone help us and narrow the exact filter/rule that make this happen? So we can change our ui to work around this :)

**Comment by MorrisJobke** [comment]
> Dammit 😁
> Can someone help us and narrow the exact filter/rule that make this happen? So we can change our ui to work around this :)

There you go: https://github.com/nextcloud/server/issues/11450#issuecomment-425717206

**Comment by skjnldsv** [comment]
Duplicate of #11450

**Comment by blizzz** [comment]
@jancborchardt https://github.com/nextcloud/server/issues/10434#issuecomment-408965731 sounds mostly like a config thing with ad blockers. So, configuring them it is imho. Default ublock origin does not do it, thus I think there's nothing we should/need do about it. Nor can sustainably. Perhaps a hint in FAQs or documentation.

**Comment by integras** [comment]
I have Ad Blocker Plus (ABP) and Ghostery - same problem.

But I agree with @blizzz : this is more a config issue to me, blockers can be disabled for such sites, at least if you use it as your own installation. I had looked in the manual, but not found any information:  

https://docs.nextcloud.com/server/14/admin_manual/configuration_files/file_sharing_configuration.html

Just a small comment regarding this issue would have been sufficient to put me right :) So if someone can just add a sentence like:

"It is known that different ad-blockers block the share link from being displayed. If this happens to you, you have the option of disabling the ad-blocker for the site of the nextcloud installation. Most ad-blockers offer this possibility directly if you click on their logo in the browser window. Do not forget to reload the page."

Thanks!

**Comment by skjnldsv** [comment]
Fixed in #11844

**Comment by SMichel72** [comment]
As I do not expect advertisments on my Nextcloud installation I've now deaktivated Adblock Plus. And it works fine again!

**Comment by dszego** [comment]
I can confirm - the share link doesn't show up right with AdBlockPlus enabled. Set an exception rule for your NextCloud site.

**Comment by noisyle** [comment]
AdBlock inject "display: none !important;" to class "share-menu", which makes the share link not visible.
![image](https://user-images.githubusercontent.com/4177157/53580521-c7949180-3bb6-11e9-8932-c6a4ebb2e6fd.png)

**Comment by MorrisJobke** [comment]
> AdBlock inject "display: none !important;" to class "share-menu", which makes the share link not visible.

Thinking about documentation in #14616

**Comment by tloubrieu** [comment]
My colleague has the same issue with the Adblocker for Chrome (NoAds) on macOS.
The  nextcloud server  version is 14.0.10, 
Could you re-open this issue ?

**Comment by AxelTheGerman** [comment]
I was searching for the "share link" option for at least 5 minutes before decided to search for open issues.

Any chance we could just change the CSS class name to something nextcloud specific? e.g. `nextcloud-share-menu`

I agree that working around ad blockers sucks, but having to configure exceptions everywhere also sucks.

I doubt any ad blocker would specifically target nextcloud as long as we don't use the most obviously blocked class names

**Comment by pooh22** [comment]
I've had this issue for ages as well, I didn't know why and didn't look too hard to find out. I'm usually using firefox with ublock origin and privacy badger, canvas blocker, https everywhere plugins. My nextcloud version is 18.0.6, but this has been the case since at least nextcloud 14, maybe earlier.

The problem is that when you click on the share icon, there's no indication that something is missing, it's like the sharing capability doesn't exist. Also there's no alternative route to sharing a file, so you can't work around it, other then disabling ublock origin or  using a different browser without the plugin.

This is a very basic and essential feature of nextcloud and it should just work, regardless of the plugins that on most sites protect you from trackers and ads.
